LoggerLib/Loggers/CustomLogger.cs

```
@@ -0,0 +1,56 @@
+﻿using System;
+using LoggerLib.Enums;
+
+namespace LoggerLib.Loggers
+{
+    internal class CustomLogger : BaseLogger
+    {
+        private readonly Action<LogLevel, string> log;
+
+        public CustomLogger(Action<LogLevel, string> loggerHandler, LogLevel loggerLogLevel = LogLevel.Info) 
+            : base(loggerLogLevel)
+        {
+            log = loggerHandler;
+        }
+
+        public override void Error(string message)
+        {
+            if (!HasAppropriateLevel(LogLevel.Error))
+            {
+                return;
+            }
+
+            log?.Invoke(LogLevel.Error, message);
```

I think we shouldn't check loggerHandler for `null`, it's unexpected that Custom logger has no valid loggerHandler and such error should throw an exception (the system null reference exception is enough but if you want you could validate the handler in the constructor and throw an error there).

---

Fixed